lib/benchpark/experiment.py

```diff
@@ -248,18 +279,47 @@ def compute_applications_section_wrapper(self):
 
         for cls in self.helpers:
             variables, env_vars = cls.compute_config_variables_wrapper()
-            self.variables |= variables
+            self.expr_vars.extend(variables)
             self.env_vars["set"] |= env_vars["set"]
             self.env_vars["append"][0] |= env_vars["append"][0]
 
+        self.initialize_experiment_variables()
+
+        visited_required_variables_setters = set()
+        for cls in self.__class__.mro():
+            if hasattr(cls, "register_required_variables") and cls.register_required_variables not in visited_required_variables_setters:
+                visited_required_variables_setters.add(cls.register_required_variables)
+                cls.register_required_variables(self)
```

Comment:
@scheibelp This code checks for the implementation of the `register_required_variables` method in the current class or any of its super classes in the mro hierarchy. It also makes sure the same method is not called twice under inheritance.